### PR TITLE
Update SigninBruteForce-AzurePortal.yaml

### PR DESCRIPTION
**Pull request by praveenthepro**
Modified the logic to use the dynamic thresholds rather than fixed thresholds which is causing issues some of the customer environments

   Required items, please complete
   
   Change(s):
   - Modified the logic to use the dynamic thresholds rather than fixed thresholds which is causing issues some of the customer environments 

   Reason for Change(s):
   - Modified the logic to use the dynamic thresholds rather than fixed thresholds which is causing issues some of the customer environments 

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: Solutions/Azure Active Directory/Analytic Rules/SigninBruteForce-AzurePortal.yaml [hunk 1]
```diff
@@ -22,48 +21,64 @@ tactics:
...
+    | summarize FailureCountBeforeSuccess = sumif(FailureOrSuccessCount, FailureOrSuccess == "Failure"), StartTime = min(TimeGenerated), EndTime = max(TimeGenerated), makelist(FailureOrSuccess), IPAddress = make_set(IPAddress, 15), make_set(Browser, 15), make_set(City, 15), make_set(State, 15), make_set(Region, 15), make_set(ResultType, 15) by SessionStartedUtc, UserPrincipalName, CorrelationId, AppDisplayName, UserId, Type
+    // Filter records where "Success" occurs in the middle of a session
+    | where array_index_of(list_FailureOrSuccess, "Success") != 0
+    | where array_index_of(list_FailureOrSuccess, "Success") == array_length(list_FailureOrSuccess) - 1
+    // Remove unnecessary columns from the output
+    | project-away SessionStartedUtc, list_FailureOrSuccess
+    // Join with another table and calculate deviation
+    | join kind=inner (
+        table(tableName)
+        | where TimeGenerated > ago(14d)
```
Comment (petebryan) [inline Solutions/Azure Active Directory/Analytic Rules/SigninBruteForce-AzurePortal.yaml:64]: This doesn't line up with the 7d window you set in the metadata

File: Solutions/Azure Active Directory/Analytic Rules/SigninBruteForce-AzurePortal.yaml [hunk 2]
```diff
@@ -1,9 +1,8 @@
 id: 28b42356-45af-40a6-a0b4-a554cdfd5d8a
 name: Brute force attack against Azure Portal
-description: |
-  'Identifies evidence of brute force activity against Azure Portal by highlighting multiple authentication failures and by a successful authentication within a given time window. 
-  Default Failure count is 10 and default Time Window is 20 minutes.
-  References: https://docs.microsoft.com/azure/active-directory/reports-monitoring/reference-sign-ins-error-codes.'
+description: >
+  Detects Azure Portal brute force attacks by monitoring for multiple authentication failures and a successful login within a 20-minute window. Default settings: 10 failures, 25 deviations.
```
Comment (petebryan) [inline Solutions/Azure Active Directory/Analytic Rules/SigninBruteForce-AzurePortal.yaml:4]: Are we still defaulting to 10 failures?

File: Solutions/Azure Active Directory/Analytic Rules/SigninBruteForce-AzurePortal.yaml [hunk 3]
```diff
@@ -76,5 +91,5 @@ entityMappings:
     fieldMappings:
```
Comment (petebryan) [inline Solutions/Azure Active Directory/Analytic Rules/SigninBruteForce-AzurePortal.yaml:91]: Can we add AAD User ID as an entity as well please.
Also is it possible to get a device entity included in this?

**Review by petebryan — CHANGES_REQUESTED** [review]

File: Solutions/Azure Active Directory/Analytic Rules/SigninBruteForce-AzurePortal.yaml (hunk 3 above)
Comment (praveenthepro) [reply Solutions/Azure Active Directory/Analytic Rules/SigninBruteForce-AzurePortal.yaml:93]: Added the AAD User ID as an entity also, Kindly please checkout

**Review by petebryan — APPROVED** [review]

**Pull request merged**